File: programmers/lv2/기출문제/두 큐 합 같게 만들기.py
from collections import deque

def solution(queue1, queue2):
    answer = 0
    
    q_len = len(queue1)
    q1, q2 = deque(queue1), deque(queue2)
    diff = 0
    for i in range(q_len):
        diff += q1[i] - q2[i]
    
    if diff % 2 == 1:
        return -1
    
    while not diff == 0:
        if diff > 0:
            item = q1.popleft()
            q2.append(item)
            diff -= item * 2
        else:
            item = q2.popleft()
            q1.append(item)
            diff += item * 2
        answer += 1
        if answer >= q_len * 4:
            answer = -1
            break
    
    return answer

File: programmers/lv2/기출문제/두 큐 합 같게 만들기.js
function solution(queue1, queue2) {
    var answer = 0;
    
    const len = queue1.length * 2;
    var idx1 = 0, idx2 = 0;
    var sum1 = queue1.reduce((acc, item) => acc + item, 0),
        sum2 = queue2.reduce((acc, item) => acc + item, 0);
    while (sum1 !== sum2) {
        if (sum1 > sum2) {
            if (idx1 === len) {
                return -1;
            }
            const num = queue1[idx1++];
            sum1 -= num;
            sum2 += num;
            queue2.push(num);
        } else {
            if (idx2 === len) {
                return -1;
            }
            const num = queue2[idx2++];
            sum2 -= num;
            sum1 += num;
            queue1.push(num);
        }
        answer += 1;
    }
    
    return answer;
}
